tests/unit/db/unit_t_db_base.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python
+# Copyright (c) 2015 IBM. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+_unit_t_db_base_
+
+unit_t_db_base module - The base class for all unit tests that target a db
+
+The unit tests are set to execute by default against a CouchDB instance.
+
+In order to run the unit tests against a Cloudant instance, set the
+RUN_CLOUDANT_TESTS environment variable to something.
+
+  example: export RUN_CLOUDANT_TESTS=1
+
+Other valid environment variables:
+
+CLOUDANT_ACCOUNT: Set this to the Cloudant account that you wish to connect to.
+  - This is used for Cloudant tests only.
+  
+  example: export CLOUDANT_ACCOUNT=account
+
+DB_USER: Set this to the username to connect with.  
+  - Optional for CouchDB tests.  If omitted then a user will be created before
+    tests are executed in CouchDB.
+  - Mandatory for Cloudant tests.  
+
+  example: export DB_USER=user
+
+DB_PASSWORD: Set this to the password for the user specified.
+
+  example: export CLOUDANT_PASSWORD=password
+
+DB_URL: Optionally set this to override the construction of the database URL.
+
+  example: export CLOUDANT_URL=https://account.cloudant.com
+
+"""
+
+import unittest
+import requests
+import os
+import uuid
+
+from cloudant.account import CouchDB, Cloudant
+
+class UnitTestDbBase(unittest.TestCase):
+    """
+    The base class for all unit tests targeting a database
+    """
+
+    @classmethod
+    def setUpClass(self):
+        """
+        If targeting CouchDB, Set up a CouchDB instance otherwise do nothing.
+          
+        Note: Admin Party is currently unsupported so we must create a 
+          CouchDB user for tests to function with a CouchDB instance if one is
+          not provided.
+        """
```

Comment:
ConfigParser is pretty standard in python: https://docs.python.org/2/library/configparser.html. Can fix it later.